client/modules/IDE/components/PreviewFrame.jsx

```diff
@@ -350,4 +359,38 @@ PreviewFrame.defaultProps = {
   cmController: {}
 };
 
-export default PreviewFrame;
+const mapStateToProps = state => ({
+  files: state.files,
+  htmlFile: getHTMLFile(state.files),
+  content:
+    (state.files.find(file => file.isSelectedFile) ||
+    state.files.find(file => file.name === 'sketch.js') ||
+    state.files.find(file => file.name !== 'root')).content,
+  isPlaying: state.ide.isPlaying,
+  isAccessibleOutputPlaying: state.ide.isAccessibleOutputPlaying,
+  previewIsRefreshing: state.ide.previewIsRefreshing,
+  textOutput: state.preferences.textOutput,
+  gridOutput: state.preferences.gridOutput,
+  soundOutput: state.preferences.soundOutput,
+  language: state.preferences.language,
+  autorefresh: state.preferences.autorefresh,
+});
+
+
+const mapDispatchToProps = dispatch => bindActionCreators(
+  Object.assign(
```

Comment:
This isn't right, as these functions aren't getting passed to the `PreviewFrame` component and the app crashes when you click play. You shouldn't need to use `Object.assign` here.

Reply:
ok. I transformed mapDispatchToProps to an object. the tests pass and the editor seems to run now

Reply:
working for me now! thank you ✨